### PR TITLE
[IA-3716] Fix v2 list

### PR DESCRIPTION
**Pull request by jdcanas**
So the issue with runtimes showing up due via creator filtering is partially an issue with the V2 list endpoint. The list function in v2 doesn't perform the fallback properly to handle runtimes without

Proof in curl below (the auth token is expired, note the fact its the same request, but the v1 endpoint returns more runtimes than the v2). This makes a UI running against local leo work properly for v1 runtimes. 

**Still need to test the creator filtering for runtimes created with v2 endpoint.** Want to socialize this fix first though.

image

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala [hunk 1]
```diff
@@ -352,10 +354,21 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
         .map(_.getOrElse(List.empty))
 
       // We must also check the RuntimeSamResourceId in sam to support already existing and newly created google runtimes
-      samVisibleRuntimeSamResourceIds <- NonEmptyList
-        .fromList(runtimesUserIsNotCreator.map(_.samResource))
-        .traverse(ids => authProvider.filterUserVisible(ids, userInfo))
-        .map(_.getOrElse(List.empty))
+      runtimesAndProjects = runtimesUserIsNotCreator.map(r => (GoogleProject(r.cloudContext.asString), r.samResource))
```
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala]: `(GoogleProject(r.cloudContext.asString)` doesn't seem right cuz you the cloud context could be either GCP or Azure

Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala]: This is the fallback for runtimes created with the v1 endpoint. the `GoogleProject` cast is for compatibility with the `authProvider`

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala]: this is not fallback for just runtimes created via v1 api tho..runtimes created via v2 will show up here too

**Comment by Qi77Qi** [comment]
closing this PR since it's superceded by https://github.com/DataBiosphere/leonardo/pull/3061

**Pull request closed without merging**